kasa/discover.py

```diff
@@ -259,10 +283,11 @@ async def discover(
     async def discover_single(
         host: str,
         *,
+        discovery_timeout: int = 5,
         port: Optional[int] = None,
-        timeout=5,
+        timeout: Optional[int] = None,
```

Comment:
The reason for adding a separate timeout (and renaming the existing one) is unclear to me. Maybe this change should be extracted also into its own PR so we can discuss it without derailing this PR?

Reply:
It's because discovery creates the connection parameters with a timeout for the device transports over TCP.  This could and should be different to the actual discovery timeout over UDP which may be a bigger value to listen for all devices on the network.  This is consistent with the CLI which has separate discovery and device timeout parameters.  I think this is an intrinsic part of this PR because it's here we are having discovery create connection params for future connections.  Make sense?